frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -870,4 +926,51 @@ mod tests {
 			assert!(matches!(call, OuterCall::MultiPhase(Call::submit_unsigned(_, _))));
 		})
 	}
+
+	#[test]
+	fn trim_compact_for_length_does_not_modify_when_short_enough() {
```

Comment:
I think these two tests can be combined where you basically test 

1. `encoded_len > compact_len`
2. `encoded_len == compact_len`
3.`encoded_len < compact_len`

Reply:
I think case 2 and 3 there are kind of redundant, given that what we're ultimately checking is that `encoded_len <= max_allowed_len`. We could definitely combine the two tests, which map to 1 and 2 respectively; I separated them so it's clear from the results which direction failed.

Reply:
fair enough